types/stripe/index.d.ts

```diff
@@ -13900,9 +13900,16 @@ declare namespace Stripe {
             };
             readonly requestId: string;
             readonly detail?: any;
-            readonly: number;
             readonly params?: string;
             readonly type: string;
+            readonly statusCode?: number;
+
+            readonly charge?: charges.ICharge;
```

Comment:
When I see this in the response JSON it's a string Charge ID.  Is this dependent upon some expand parameter?  Should it be `string | charges.ICharge`?

Does this actually get put on the IStripeError?  In my testing I see it in `raw` instead.

```javascript
stripe.charges.create(
  {amount: 1000, currency: "usd", source: "tok_chargeDeclinedFraudulent"},
  function(err, charges) {
    console.error(JSON.stringify(err, null, 2));
    console.log(charges);
  }
);

{
  "type": "StripeCardError",
  "stack": "<snip>",
  "rawType": "card_error",
  "code": "card_declined",
  "message": "Your card was declined.",
  "raw": {
    "charge": "ch_1FRllIBCvBiGc7Sd5lEOpjTG",
    "code": "card_declined",
    "decline_code": "fraudulent",
    "doc_url": "https://stripe.com/docs/error-codes/card-declined",
    "message": "Your card was declined.",
    "type": "card_error",
    "headers": {
      "server": "nginx",
      "date": "Wed, 09 Oct 2019 20:27:56 GMT",
      "content-type": "application/json",
      "content-length": "264",
      "connection": "keep-alive",
      "access-control-allow-credentials": "true",
      "access-control-allow-methods": "GET, POST, HEAD, OPTIONS, DELETE",
      "access-control-allow-origin": "*",
      "access-control-expose-headers": "Request-Id, Stripe-Manage-Version, X-Stripe-External-Auth-Required, X-Stripe-Privileged-Session-Required",
      "access-control-max-age": "300",
      "cache-control": "no-cache, no-store",
      "request-id": "req_ocVR3raJUZZPZA",
      "stripe-version": "2019-08-14",
      "strict-transport-security": "max-age=31556926; includeSubDomains; preload"
    },
    "statusCode": 402,
    "requestId": "req_ocVR3raJUZZPZA"
  },
  "headers": {
    "server": "nginx",
    "date": "Wed, 09 Oct 2019 20:27:56 GMT",
    "content-type": "application/json",
    "content-length": "264",
    "connection": "keep-alive",
    "access-control-allow-credentials": "true",
    "access-control-allow-methods": "GET, POST, HEAD, OPTIONS, DELETE",
    "access-control-allow-origin": "*",
    "access-control-expose-headers": "Request-Id, Stripe-Manage-Version, X-Stripe-External-Auth-Required, X-Stripe-Privileged-Session-Required",
    "access-control-max-age": "300",
    "cache-control": "no-cache, no-store",
    "request-id": "req_ocVR3raJUZZPZA",
    "stripe-version": "2019-08-14",
    "strict-transport-security": "max-age=31556926; includeSubDomains; preload"
  },
  "requestId": "req_ocVR3raJUZZPZA",
  "statusCode": 402
}
```

Reply:
Good catch @pushplay. I've only tested this with payment intents where we do get the whole Payment Intent object, so I made the assumption that the same applies to the other properties. The following docs has the correct types, I'll update the PR accordingly: https://stripe.com/docs/api/errors

Reply:
Regarding if the property is actually put on the StripeError, I've based the change on this commit on version 7.10.0 of the SDK (release recently): https://github.com/stripe/stripe-node/pull/699/commits/04c514ffef079edbb345e4e7985ba074c6783a93#diff-872c0c131ce19fb5a074d972c85c835dR40

I tested the same code, and got the following response:

```
{
  "raw": {
    "charge": "ch_1FS0iqFuhr4V1leg9qY4ygvN",
    "code": "card_declined",
    "decline_code": "fraudulent",
    "doc_url": "https://stripe.com/docs/error-codes/card-declined",
    "message": "Your card was declined.",
    "type": "card_error",
    "headers": {
      "server": "nginx",
      "date": "Thu, 10 Oct 2019 12:26:25 GMT",
      "content-type": "application/json",
      "content-length": "264",
      "connection": "keep-alive",
      "access-control-allow-credentials": "true",
      "access-control-allow-methods": "GET, POST, HEAD, OPTIONS, DELETE",
      "access-control-allow-origin": "*",
      "access-control-expose-headers": "Request-Id, Stripe-Manage-Version, X-Stripe-External-Auth-Required, X-Stripe-Privileged-Session-Required",
      "access-control-max-age": "300",
      "cache-control": "no-cache, no-store",
      "idempotency-key": "stripe-node-retry-326def3b-ce89-457e-a138-380a8ab25d92",
      "original-request": "req_MMKmgvh6srfYAy",
      "request-id": "req_MMKmgvh6srfYAy",
      "stripe-should-retry": "false",
      "stripe-version": "2019-05-16",
      "strict-transport-security": "max-age=31556926; includeSubDomains; preload"
    },
    "statusCode": 402,
    "requestId": "req_MMKmgvh6srfYAy"
  },
  "rawType": "card_error",
  "code": "card_declined",
  "headers": {
    "server": "nginx",
    "date": "Thu, 10 Oct 2019 12:26:25 GMT",
    "content-type": "application/json",
    "content-length": "264",
    "connection": "keep-alive",
    "access-control-allow-credentials": "true",
    "access-control-allow-methods": "GET, POST, HEAD, OPTIONS, DELETE",
    "access-control-allow-origin": "*",
    "access-control-expose-headers": "Request-Id, Stripe-Manage-Version, X-Stripe-External-Auth-Required, X-Stripe-Privileged-Session-Required",
    "access-control-max-age": "300",
    "cache-control": "no-cache, no-store",
    "idempotency-key": "stripe-node-retry-326def3b-ce89-457e-a138-380a8ab25d92",
    "original-request": "req_MMKmgvh6srfYAy",
    "request-id": "req_MMKmgvh6srfYAy",
    "stripe-should-retry": "false",
    "stripe-version": "2019-05-16",
    "strict-transport-security": "max-age=31556926; includeSubDomains; preload"
  },
  "requestId": "req_MMKmgvh6srfYAy",
  "statusCode": 402,
  "charge": "ch_1FS0iqFuhr4V1leg9qY4ygvN",
  "decline_code": "fraudulent"
}
```

Reply:
Oh that charge bit is brand new then.  I'm on 7.9.0.

I think we should update the version in the header then.  I'm pretty sure we've already been incorporating changes from v7.

Reply:
Sure, I can do that as part of this PR if you want?

Reply:
Yes please.

Reply:
Done 👍